presto-spi/src/main/java/com/facebook/presto/spi/memory/ArrayPool.java

```diff
@@ -16,18 +16,60 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
 
-public class ByteArrayPool
+import static java.util.Objects.requireNonNull;
+
+public class ArrayPool<ArrayType>
 {
+    public static abstract class Allocator<ArrayType>
+    {
+        abstract ArrayType alloc(int size);
+        abstract void init(ArrayType array);
+        abstract int getLength(ArrayType array);
+    }
+
+    private static class LongArrayList
```

Comment:
This class is used for `times`, but `times` itself seems to be used only in `trim` method which doesn't seem to be doing anything. Could you explain how it works?

Reply:
The trim() function was not finished. I'll finish it in the next iteration.